crates/relayer/src/foreign_client.rs

```diff
@@ -1359,6 +1355,14 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
                 ForeignClientError::client_query(self.id().clone(), self.src_chain.id(), e)
             })?;
 
+        // This is necessary because the results are sorted in lexicographic order instead of
+        // numeric order, and we cannot therefore rely on setting the `reverse = true` in the
+        // `PageRequest` setting. Since we are asking for all states anyway, we can sort them
+        // by height ourselves.
```

Comment:
Should we remove `fetch_consensus_states()`? Doesn't seem to be used anymore.

Reply:
It's still used from the CLI, but as you note below we can simplify it by only listing the heights

Reply:
Good catch!

Reply:
87d9692ded25c6b772a8e5ef1b37b8b1e0a09565